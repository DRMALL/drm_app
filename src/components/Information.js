import React, { Component } from 'react'
import { View, Text, Image, TouchableOpacity, StyleSheet } from 'react-native'
import { StackNavigator } from 'react-navigation'
import { primaryColor, loginBackgroundColor } from '../common/constants'
import { basicDocument, userName, companyName, phoneNumber, postalAddress, securitySetting, resetPassword, personalInformation } from '../common/strings'
import { information } from '../styles'
import TouchIntoText from './units/TouchIntoText'

const gobackWhiteIcon = require('../images/navigation_icons/goback_white.png')
const emptyIcon = require('../images/navigation_icons/empty.png')

export default class Information extends Component {
  static navigationOptions = ({ navigation })=> ({
    headerStyle: {
      backgroundColor: primaryColor,
    },
    headerTitle: <Text style={{ fontSize: 20, color: '#FFF', alignSelf: 'center' }} >{personalInformation}</Text>,
    headerLeft: <TouchableOpacity style={{padding: 10, paddingLeft: 20}} onPress={() => navigation.goBack()}>
      <Image source={gobackWhiteIcon}/>
    </TouchableOpacity>,
    headerRight: <Image style={{marginLeft: 20}} source={emptyIcon}/>,
  })
  render() {
    return (
      <View style={information.wrap}>
        <Text style={information.text} >{basicDocument}</Text>
        <TouchIntoText title={userName} value={'戴永明'} activeOpacity={0.6} onPress={() => this.props.navigation.navigate('username', {name: 'UserName'})} />
        <TouchIntoText title={companyName} value={'公司名称公司名称公司名称公司名称'} activeOpacity={0.6} onPress={() => this.props.navigation.navigate('companyname', {name: 'CompanyName'})} />
        <TouchIntoText title={phoneNumber} value={'18888888888'} activeOpacity={0.6} onPress={() => this.props.navigation.navigate('phone', {name: 'Phone'})} />
        <TouchIntoText title={postalAddress} value={'通讯地址通讯地址通讯'} activeOpacity={0.6} onPress={() => this.props.navigation.navigate('address', {name: 'Address'})} />
        <Text style={information.text} >{securitySetting}</Text>
        <TouchIntoText title={resetPassword} activeOpacity={0.6} onPress={() => this.props.navigation.navigate('resetpassword', {name: 'ResetPassword'})} />
      </View>
    )
  }
}