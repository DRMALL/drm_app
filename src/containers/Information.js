import React, { Component } from 'react'
import { View, Text, Image, TouchableOpacity, Alert } from 'react-native'
import { NavigationActions } from 'react-navigation'
import { primaryColor, loginBackgroundColor } from '../common/constants'
import { basicDocument, 
        userName, 
        companyName, 
        phoneNumber, 
        postalAddress, 
        securitySetting, 
        resetPassword, 
        personalInformation, 
        tokenKey,
        logOut,
        internalServerError,
} from '../common/strings'
import { information } from '../styles'
import TouchIntoText from '../components/units/TouchIntoText'
import Button from '../components/units/Button'
import { checkToken, depositToken, clearToken } from '../utils/handleToken'
import { getPort } from '../utils/fetchMethod'
import { getInfo } from '../apis'

import store from '../utils/store'
import infoAC from '../actions/infoAC'
import getInformation from '../funcs/information/getInformation'
import confirmLogOut from '../funcs/information/confirmLogOut'
import pressLogOut from '../funcs/information/pressLogOut'

const gobackWhiteIcon = require('../images/navigation_icons/goback_white.png')
const emptyIcon = require('../images/navigation_icons/empty.png')

export default class Information extends Component {
  static navigationOptions = ({ navigation })=> ({
    headerStyle: {
      backgroundColor: primaryColor,
    },
    headerTitle: <Text style={{ fontSize: 20, color: '#FFF', alignSelf: 'center' }} >{personalInformation}</Text>,
    headerLeft: <TouchableOpacity style={{padding: 10, paddingLeft: 20}} onPress={() => navigation.navigate('main')}>
      <Image source={gobackWhiteIcon}/>
    </TouchableOpacity>,
    headerRight: <Image style={{marginLeft: 20}} source={emptyIcon}/>,
  });

  constructor(props) {
    super(props)
    this.state = store.getState().information
  }
  componentDidMount() {
    getInformation()
  }

  componentWillMount() {
    this.unsubscribe = store.subscribe( ()=> this.setState(store.getState().information) )
  }

  componentWillUnmount(){
    this.unsubscribe()
  }

  render() {
    let { user_name, company_name, phone_number, postal_address } = this.state
      , { navigation } = this.props
    return (
      <View style={information.wrap}>
        <Text style={information.text} >{basicDocument}</Text>
        <TouchIntoText title={userName} 
          value={user_name === 'null' ? '' : user_name} 
          activeOpacity={0.6} 
          onPress={() => navigation.navigate('username', {user_name: user_name})} 
        />
        <TouchIntoText title={companyName} 
          value={company_name === 'null' || company_name === 'undefined' ? '' : company_name} 
          activeOpacity={0.6} 
          onPress={() => navigation.navigate('companyname', {company_name: company_name})} 
        />
        <TouchIntoText title={phoneNumber} 
          value={phone_number} 
          activeOpacity={0.6} 
          onPress={() => navigation.navigate('phone', {phone_number: `${phone_number}`})} 
        />
        <TouchIntoText title={postalAddress} 
          value={postal_address === 'null' || postal_address === 'undefined' ? '' : postal_address} 
          activeOpacity={0.6} 
          onPress={() => navigation.navigate('address', {postal_address: postal_address})} 
        />
        <Text style={information.text} >{securitySetting}</Text>
        <TouchIntoText title={resetPassword} 
          activeOpacity={0.6} 
          onPress={() => navigation.navigate('resetpassword', {name: 'ResetPassword'})} 
        />
        <View style={{alignItems: 'center'}}>
          <Button title={logOut} 
            titleStyle={information.buttonText} 
            style={information.buttonView} 
            activeOpacity={0.8}
            onPress={()=> pressLogOut(this.props)}
          />
        </View>
      </View>
    )
  }
}