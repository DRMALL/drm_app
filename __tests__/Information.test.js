import 'react-native'
import React from 'react'
import Information from '../src/containers/Information'
import renderer from 'react-test-renderer';

test('Information test', () => {
  const tree = renderer.create(
    <Information />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})