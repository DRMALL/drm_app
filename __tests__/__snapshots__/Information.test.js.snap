// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Information test 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F7F7F7",
      "height": "100%",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "borderColor": "rgba(0,0,0,.06)",
        "borderWidth": 0.5,
        "color": "rgba(0,0,0,.3)",
        "fontSize": 16,
        "height": 40,
        "paddingLeft": 15,
        "paddingVertical": 11,
      }
    }
  >
    基本资料
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "position": "relative",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderColor": "rgba(0,0,0,.06)",
          "borderWidth": 0.5,
          "flexDirection": "row",
          "height": 60,
          "opacity": 1,
          "width": "100%",
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 18,
            "left": 15,
            "opacity": 0.8,
            "position": "absolute",
            "textAlign": "left",
          }
        }
      >
        用户名
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={2}
        style={
          Object {
            "color": "rgba(0,0,0,.3)",
            "fontSize": 16,
            "position": "absolute",
            "right": 40,
            "textAlign": "right",
            "width": 180,
          }
        }
      >
        
      </Text>
      <Image
        source={1}
        style={
          Object {
            "position": "absolute",
            "right": 15,
            "tintColor": "rgba(0,0,0,.8)",
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "position": "relative",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderColor": "rgba(0,0,0,.06)",
          "borderWidth": 0.5,
          "flexDirection": "row",
          "height": 60,
          "opacity": 1,
          "width": "100%",
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 18,
            "left": 15,
            "opacity": 0.8,
            "position": "absolute",
            "textAlign": "left",
          }
        }
      >
        公司名称
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={2}
        style={
          Object {
            "color": "rgba(0,0,0,.3)",
            "fontSize": 16,
            "position": "absolute",
            "right": 40,
            "textAlign": "right",
            "width": 180,
          }
        }
      >
        
      </Text>
      <Image
        source={1}
        style={
          Object {
            "position": "absolute",
            "right": 15,
            "tintColor": "rgba(0,0,0,.8)",
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "position": "relative",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderColor": "rgba(0,0,0,.06)",
          "borderWidth": 0.5,
          "flexDirection": "row",
          "height": 60,
          "opacity": 1,
          "width": "100%",
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 18,
            "left": 15,
            "opacity": 0.8,
            "position": "absolute",
            "textAlign": "left",
          }
        }
      >
        联系电话
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={2}
        style={
          Object {
            "color": "rgba(0,0,0,.3)",
            "fontSize": 16,
            "position": "absolute",
            "right": 40,
            "textAlign": "right",
            "width": 180,
          }
        }
      >
        
      </Text>
      <Image
        source={1}
        style={
          Object {
            "position": "absolute",
            "right": 15,
            "tintColor": "rgba(0,0,0,.8)",
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "position": "relative",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderColor": "rgba(0,0,0,.06)",
          "borderWidth": 0.5,
          "flexDirection": "row",
          "height": 60,
          "opacity": 1,
          "width": "100%",
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 18,
            "left": 15,
            "opacity": 0.8,
            "position": "absolute",
            "textAlign": "left",
          }
        }
      >
        通讯地址
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={2}
        style={
          Object {
            "color": "rgba(0,0,0,.3)",
            "fontSize": 16,
            "position": "absolute",
            "right": 40,
            "textAlign": "right",
            "width": 180,
          }
        }
      >
        
      </Text>
      <Image
        source={1}
        style={
          Object {
            "position": "absolute",
            "right": 15,
            "tintColor": "rgba(0,0,0,.8)",
          }
        }
      />
    </View>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "borderColor": "rgba(0,0,0,.06)",
        "borderWidth": 0.5,
        "color": "rgba(0,0,0,.3)",
        "fontSize": 16,
        "height": 40,
        "paddingLeft": 15,
        "paddingVertical": 11,
      }
    }
  >
    安全设置
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "position": "relative",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderColor": "rgba(0,0,0,.06)",
          "borderWidth": 0.5,
          "flexDirection": "row",
          "height": 60,
          "opacity": 1,
          "width": "100%",
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 18,
            "left": 15,
            "opacity": 0.8,
            "position": "absolute",
            "textAlign": "left",
          }
        }
      >
        修改密码
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={2}
        style={
          Object {
            "color": "rgba(0,0,0,.3)",
            "fontSize": 16,
            "position": "absolute",
            "right": 40,
            "textAlign": "right",
            "width": 180,
          }
        }
      />
      <Image
        source={1}
        style={
          Object {
            "position": "absolute",
            "right": 15,
            "tintColor": "rgba(0,0,0,.8)",
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderColor": "#fff",
          "borderRadius": 10,
          "borderWidth": 0.5,
          "justifyContent": "center",
          "marginTop": 30,
          "opacity": 1,
          "paddingHorizontal": 120,
          "paddingVertical": 15,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#4DADF7",
            "fontSize": 18,
          }
        }
      >
        退出登录
      </Text>
    </View>
  </View>
</View>
`;
